node/self_out_response.js

```diff
@@ -71,9 +71,13 @@ function passResponse(args, isLast ) {
         self.inres.code = self.code;
         self.inres.ok = self.ok;
         self.first = false;
-        self.inreq.responseEvent.emit(self, self.inres);
+        process.nextTick(emitResponse);
```

Comment:
necessary to avoid race condition; similar to nextTick in error path.

otherwise we emit `finish` on the outres before the handler completes

```js
function (req, res, arg2, arg3) {
  res.sendOk();
  console.log('please first');
  console.log('why am I second :(');
}

.request().send(..., function onResponse(err, res) {
  console.log('out res is actually first lol; zalgo lol);
})
```